src/kvstore/wal/FileBasedWal.cpp

```diff
@@ -283,20 +283,33 @@ void FileBasedWal::closeCurrFile() {
         return;
     }
 
-    CHECK_EQ(fsync(currFd_), 0) << strerror(errno);
+    bool synced = false;
+    if (lastLogId_ > firstLogId_) {
```

Comment:
Why not fsync every time?
BTW, I think if fsync failed, maybe crash would be better, WDYT?